subprojects/gradle/module-dependencies-graph/src/gradleTest/kotlin/com/avito/module/dependencies/FindAndroidAppTaskActionTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.avito.module.dependencies
+
+import com.avito.logger.StubLoggerFactory
+import com.avito.module.configurations.ConfigurationType
+import com.avito.module.internal.dependencies.AndroidAppsGraphBuilder
+import com.avito.module.internal.dependencies.DependenciesGraphBuilder
+import com.avito.module.internal.dependencies.FindAndroidAppTaskAction
+import com.avito.module.internal.dependencies.FindAndroidAppTaskAction.Verdict.NoSuitableApps
+import com.google.common.truth.Truth.assertThat
+import org.gradle.util.Path.path
+import org.junit.jupiter.api.Test
+
+public class FindAndroidAppTaskActionTest {
+
+    private val root = rootProject()
+    private val graphBuilder = AndroidAppsGraphBuilder(DependenciesGraphBuilder(root, StubLoggerFactory))
+    private val action = FindAndroidAppTaskAction(graphBuilder)
+
+    @Test
+    public fun `NoSuitableApps - Result contains all no suitable apps`() {
+        val rootApp = androidApp("rootApp", root)
+        val rootAppSecond = androidApp("rootAppSecond", root)
+        val a = javaLib("a", rootApp)
```

Comment:
Can we use an Android application as a parent Gradle project?
I have doubts that it will work properly in a real project.

Reply:
In that case, this is not necessary because we don't build a graph. We need only project objects
Could you describe your doubts and maybe we need a special integration test

Reply:
![Screenshot from 2021-03-24 12-33-48](https://user-images.githubusercontent.com/1104540/112287249-3b71cd00-8c9d-11eb-94ac-d725bdaf1cda.png)

In this case, the library is inside the application. It works but looks weird. 

I suspect problems with impact analysis and build cache.
Anyway, I don't see reasons to support and test such cases.
This is why I want to avoid them everywhere.